GVRf/Framework/framework/src/main/java/org/gearvrf/scene_objects/GVRViewSceneObject.java

```diff
@@ -495,42 +592,31 @@ public void onGlobalLayout() {
                         }
                     });
 
-            mGVRContext.getActivity().registerView(RootViewGroup.this);
+            mGVRContext.getActivity().registerView(this);
         }
 
         private void onLayoutReady() {
-            /**
-             * Creates render data and material to its scene object on GL Thread.
-             */
-            mGVRContext.runOnGlThread(new Runnable() {
-                @Override
-                public void run() {
-                    createRenderData();
+            synchronized (mSceneObject.mLock) {
+                createRenderData();
 
-                    onRenderDataReady();
-                }
-            });
+                onRenderDataReady();
+
+                mSceneObject.mLock.notify();
+            }
         }
 
         private void onRenderDataReady() {
-            mViewSize = Math.max(getWidth(), getHeight());
-            mQuadWidth = getWidth() / mViewSize;
-            mQuadHeight = getHeight() / mViewSize;
-
-            /**
-             * Makes the view visible to call draw(...) and start rendering it.
-             */
-            mGVRContext.getActivity().runOnUiThread(new Runnable() {
+            mGVRContext.runOnGlThread(new Runnable() {
                 @Override
                 public void run() {
-                    // To call draw(...) after renderData has been created
-                    setVisibility(VISIBLE);
-
-                    onViewVisible();
+                    mSceneObject.onStartRendering();
```

Comment:
Can you please deliver the callback to the framework thread instead? I would like not to promote access to the rendering thread. In the actual implementation of onStartRendering you can runOnGlThread if it is necessary.